app.js

```diff
@@ -0,0 +1,34 @@
+let allBooks = document.querySelector('.books');
+
+document.getElementById('btn').addEventListener('click' , getBooks);
+
+
+function getBooks(){
+    // using Fetch 
+    fetch('books.json')
+    .then(function(re){
+        return re.json();
+    })
+
+    .then(function(books){
+
+        let bookDiv = '';
+        let title = '';
+        let author = '';
+
+        books.forEach(function(book){
+          console.log(books);
+        bookDiv += `<div class='book'> ${title} ${author} </div> `;
+        title = `<div class = 'title'> ${book.title}</div> ` ;
```

Comment:
I am pretty sure you need to define the `title `and `author `variables before using them in the `bookDiv += `expression

Reply:
I think that would solve the problem you had as well

Reply:
These two variables can only be scoped inside the callback, no need to keep them in the same scope as `bookDiv`

Reply:
Thank you Josip for responding :)
I Declared those two variables in second '. then '  but outside of the forEach loop